libs/wingc/src/lsp/signature.rs

```diff
@@ -31,88 +32,98 @@ pub fn on_signature_help(params: lsp_types::SignatureHelpParams) -> Option<Signa
+			.iter()
+			.find(|arg| arg.1.span.contains(&params.text_document_position_params.position));
+
+		let active_parameter = if named_arg_pos.is_some() {
+			sig.parameters.len() - 1
+		} else {
+			provided_args.pos_args.len() - positional_arg_pos
+		};
+
+		let param_data = sig
+			.parameters
+			.iter()
+			.map(|p| format!("{}: {}", p.name, p.typeref))
+			.collect_vec();
 
-				let active_parameter = if named_arg_pos.is_some() {
-					sig.parameters.len() - 1
-				} else {
-					provided_args.pos_args.len() - positional_arg_pos
-				};
+		let param_text = param_data.join(", ");
+		let label = format!("({}): {}", param_text, sig.return_type);
 
-				let param_data = sig
+		let signature_info = SignatureInformation {
+			label,
+			documentation: None,
+			parameters: Some(
+				sig
 					.parameters
 					.iter()
-					.map(|p| format!("{}: {}", p.name, p.typeref))
-					.collect_vec();
-
-				let param_text = param_data.join(", ");
-				let label = format!("({}): {}", param_text, sig.return_type);
-
-				let signature_info = SignatureInformation {
-					label,
-					documentation: None,
-					parameters: Some(
-						sig
-							.parameters
-							.iter()
-							.enumerate()
-							.map(|p| ParameterInformation {
-								label: ParameterLabel::Simple(
-									param_data
-										.get(p.0)
-										.unwrap_or(&format!("{}: {}", p.0, p.1.typeref))
-										.clone(),
-								),
-								documentation: if let Some(structy) = p.1.typeref.maybe_unwrap_option().as_struct() {
-									// print all fields
-									let fields = structy
-										.env
-										.iter(true)
-										.map(|f| format!("  {}: {};", f.0, f.1.as_variable().unwrap().type_))
-										.join("\n");
-									Some(lsp_types::Documentation::MarkupContent(lsp_types::MarkupContent {
-										kind: lsp_types::MarkupKind::Markdown,
-										value: format!("```wing\nstruct {} {{ \n{}\n}}\n```", structy.name.name, fields),
-									}))
-								} else {
-									None
-								},
-							})
-							.collect(),
-					),
-
-					active_parameter: Some(active_parameter as u32),
-				};
-
-				return Some(SignatureHelp {
-					signatures: vec![signature_info],
-					active_signature: None,
-					active_parameter: None,
-				});
-			}
-		}
+					.enumerate()
+					.map(|p| ParameterInformation {
+						label: ParameterLabel::Simple(
+							param_data
+								.get(p.0)
+								.unwrap_or(&format!("{}: {}", p.0, p.1.typeref))
+								.clone(),
+						),
+						documentation: if let Some(structy) = p.1.typeref.maybe_unwrap_option().as_struct() {
+							// print all fields
+							let fields = structy
+								.env
+								.iter(true)
+								.map(|f| format!("  {}: {};", f.0, f.1.as_variable().unwrap().type_))
+								.join("\n");
+							Some(lsp_types::Documentation::MarkupContent(lsp_types::MarkupContent {
+								kind: lsp_types::MarkupKind::Markdown,
+								value: format!("```wing\nstruct {} {{ \n{}\n}}\n```", structy.name.name, fields),
```

Comment:
I think it's confusing to see the struct's name. Lots of people simply don't know about the keyword argument feature because of tht.

I think we should just flatten the keyword arguments. There's no need to mention the "props" structs at all.

For example, the doc of `new cloud.Function` should look something like this:

---


```js
new cloud.Function(handler, ...);
```

### Arguments

- `handler: (event: str): str` - the inflight closure to call...

### Keyword arguments
- `timeout: duration` - duration to wait before ...
- `memory: num` - how much memory to allocate for this function...
- ...

---

What do you think?